pdns/dnsdist.cc

```diff
@@ -1850,9 +1850,12 @@ static void setUpLocalBind(std::unique_ptr<ClientState>& cs)
 
   if(!cs->tcp && IsAnyAddress(cs->local)) {
     int one=1;
-    setsockopt(fd, IPPROTO_IP, GEN_IP_PKTINFO, &one, sizeof(one));     // linux supports this, so why not - might fail on other systems
+
+    if (setsockopt(fd, IPPROTO_IP, GEN_IP_PKTINFO, &one, sizeof(one)) < 0)    // linux supports this, so why not - might fail on other systems
```

Comment:
the comment suggests that this does not warrant a warning

Reply:
Indeed, I don't think we want a warning for this.